lib/src/selection.dart

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// Where a selection start or end point may appear in some piece of text.
```

Comment:
```suggestion
/// A region of text selected by an optional start and end index.
```

Could consider calling out that the indices are for UTF-16 code units, but that's probably what most readers would assume.

Reply:
Done.